pkg/cli/iam_handle.go

```diff
@@ -0,0 +1,166 @@
+	Do(context.Context, *v1alpha1.IAMRequestWrapper) ([]*v1alpha1.IAMResponse, error)
+}
+
+// IAMHandleCommand handles IAM requests.
+type IAMHandleCommand struct {
+	cli.BaseCommand
+
+	flagPath string
+
+	flagDuration time.Duration
+
+	// testHandler is used for testing only.
+	testHandler Handler
+}
+
+func (c *IAMHandleCommand) Desc() string {
+	return `Handle the IAM request YAML file in the given path`
+}
+
+func (c *IAMHandleCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+Handle the IAM request YAML file in the given path:
+
+      aod iam handle -path "/path/to/file" -duration "2h"
+`
+}
+
+func (c *IAMHandleCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "path",
+		Target:  &c.flagPath,
+		Example: "/path/to/file",
+		Usage:   `The path of IAM request file.`,
+	})
+
+	f.DurationVar(&cli.DurationVar{
+		Name:    "duration",
+		Target:  &c.flagDuration,
+		Example: "2h",
+		Usage:   `The IAM permission lifecycle, as a duration.`,
+	})
+
+	return set
+}
+
+func (c *IAMHandleCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %v", args)
+	}
+
+	if c.flagPath == "" {
+		return fmt.Errorf("path is required")
+	}
+
+	if c.flagDuration <= 0 {
+		return fmt.Errorf("a positive duration is required")
+	}
+
+	return c.handleIAM(ctx)
+}
+
+func (c *IAMHandleCommand) handleIAM(ctx context.Context) error {
+	// Unmarshal the YAML file to IAMRequest
+	data, err := os.ReadFile(c.flagPath)
+	if err != nil {
+		return fmt.Errorf("failed to read file from %q, %w", c.flagPath, err)
+	}
+	var req *v1alpha1.IAMRequest
+	if err := yaml.Unmarshal(data, &req); err != nil {
+		return fmt.Errorf("failed to unmarshal yaml to IAMRequest: %w", err)
+	}
+
+	var h Handler
+	if c.testHandler != nil {
+		// Use testHandler is it is for testing.
+		h = c.testHandler
+	} else {
+		// Create resource manager clients.
+		organizationsClient, err := resourcemanager.NewOrganizationsClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create organizations client: %w", err)
+		}
+		defer organizationsClient.Close()
+
+		foldersClient, err := resourcemanager.NewFoldersClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create folders client: %w", err)
+		}
+		defer foldersClient.Close()
+
+		projectsClient, err := resourcemanager.NewProjectsClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create projects client: %w", err)
+		}
+		defer projectsClient.Close()
+
+		// Create IAMHandler with the clients.
+		h, err = handler.NewIAMHandler(
+			ctx,
+			organizationsClient,
+			foldersClient,
+			projectsClient,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create IAM handler: %w", err)
+		}
+	}
```

Comment:
I can't recall precisely because there could be a catch. @sqin2019 I remember you did some research earlier, right?

Reply:
The primary reason is to limit the capability, we only want org/folder/project level iam changes. On google [guide](https://cloud.google.com/iam/docs/granting-changing-revoking-access) also recommends the resource manager clients. --> "You can manage access to projects, folders, and organizations with the Google Cloud console, the Google Cloud CLI, the REST API, or the [Resource Manager client libraries](https://cloud.google.com/resource-manager/docs/libraries)."

It is also doable with the generic IAM client since we already limited the resource type in code. Let me know if we should change to this one instead.

Reply:
Let's talk offline.

Reply:
Discussed with Chen, AI: I will try the generic IAM client and see it works for our use cases, and then follow up on this.

Reply:
https://github.com/abcxyz/access-on-demand/issues/17

Reply:
@sethvargo Should I resolve or if any further concerns?

Reply:
I will close and follow up in the issue instead, thanks!